.travis.yml

```diff
@@ -94,5 +94,9 @@ matrix:
     - language: python
       python: "3.7"
       env: TOXENV=py37
-      dist: xenial # required for Python >= 3.7
       script: *1
+    - name: "Lint Python code with flake8"
+      language: python
+      python: "3.7"
+      install: pip install flake8
+      script: flake8 . --count --exclude=./.*,*/visualization/tfdv.py --select=E9,F63,F7,F82 --show-source --statistics
```

Comment:
For consistency can we either include **roc_curve.py** within this exclude statement or add `# flake8: noqa TODO` to **tfdv.py**.

Reply:
__roc_curve.py__ passes the flake8 tests.  Why would we want to move it to this exclude statement?  We only moved __tfdv.py__ to this exclude statement because its pytests are overzealous in that the complain if a comment is added to the file.

Reply:
Previously the tests failed before `# flake8: noqa TODO` was added to **roc_curve.py** as mentioned in [this comment](https://github.com/kubeflow/pipelines/pull/1721#issuecomment-523248857). I believe we could remove **tfdv.py** from the exclude statement and just add `# flake8: noqa TODO` at the top of the file (at least from my understanding of flask8) and mimic the implementation within **roc_curve.py**. It would be simpler to keep the way we ignore these files identical and new visualizations that are added can follow the same structure defined within this PR to enforce consistency.

Reply:
If we add __# flake8: noqa TODO__ to the top of __tfdv.py__ then its pytest fails.

Reply:
Ah I realize the issue now, it is due to the snapshots being out of date. We can update the snapshots if we add **# flake8: noqa TODO** to the top of **tfdv.py**.

Reply:
As a follow up
https://github.com/kubeflow/pipelines/blob/94aa0788d5523de92c803407abb470ea053421f5/backend/src/apiserver/visualization/test_exporter.py#L59-L62
can be changed to use **test.py** rather than **tfdv.py** at line 61. This would prevent the overzealous unit tests from getting in the way of future changes to **tfdv.py** while also ensuring `create_cell_from_file` works as intended. After this is taken care of I think this is ready to be merged. Thanks for all your work @cclauss!